### PR TITLE
Enhanced Contribution Guide & Read The Docs

### PR DESCRIPTION
**Pull request by pws1453**
We added some resources into the CONTRIBUTING.md and the Read The Docs. Let us know of any feedback or suggestions you may have!

### CODE REVIEW

File: README.md
Comment (gabalafou) [inline README.md:9]:
```suggestion
In March 2019, this software was not accessible. As of April 2022, significant work is still required to reach that goal.
```

File: docs/index.md
Comment (gabalafou) [inline docs/index.md:9]:
```suggestion
In March 2019, this software was not accessible. As of April 2022, significant work is still required to reach that goal.
```

**Review by gabalafou — COMMENTED** [review]
I love the work here to bring greater clarity to visitors of this repo and the Read The Docs, thank you so much!

My one big concern with this PR is that it duplicates content—for example, notice how I had to fix the word "accessable" in two places. I wonder... is there a way to configure and use Jupyter Book (which I believe is responsible for our Read The Docs site?) so as to avoid that duplication?

File: CONTRIBUTING.md
Comment (isabela-pf) [inline CONTRIBUTING.md:36]: Are these supposed to be links too?

File: README.md
Comment (isabela-pf) [inline README.md:3]: Suggested for same reason mentioned above. Technically no formal "working group" has been formed yet.

```suggestion
Welcome to the Github repository for the accessibility working group of [Project Jupyter](https://jupyter.org/). This group is being formed in early 2019. Its goal is to gather stakeholders who can actively work with the project to make Jupyter's core user-facing software accessible. In particular, the initial targets are:
```

File: docs/_toc.yml
Comment (isabela-pf) [inline docs/_toc.yml:4]: This probably does need to be its own file. Accessibility statements typically are on their own page so that people may find them quickly.

```suggestion
- file: docs/statement.md
```

File: docs/community/README.md
Comment (isabela-pf) [inline docs/community/README.md:16]:
```suggestion
- [public notes archive](docs/community/meeting-minutes/jupyterlab-accessibility-meetings/all-minutes.md)
```

File: docs/community/README.md
Comment (isabela-pf) [inline docs/community/README.md:17]: Is there a reason to add an HTML tag for a line break here? Maybe I'm just unaware of the reasoning.

```suggestion

```

**Review by isabela-pf — COMMENTED** [review]
Hooray! I'm so glad to see this PR. Thank you for thinking about these docs holistically and with a goal of helping make them more consistent. It's great work we often don't make time to do.

I also am giving a big ditto to @gabalafou's comment about repeated information. It's hard to catch accidental changes to one area and not the other, so we risk easily having conflicting or confusing information in the future. [JupyterBook's docs](https://jupyterbook.org/intro.html) might have some help?

**Comment by gabalafou** [comment]
If you're not sure what to do about duplication and you're not finding any good solutions, I think one solution would be to simply replace the redundant content in the markdown files with links to the same content on the RTD site. For example, perhaps instead of having a README.md full of content, it would just say something like: 

-----

All documentation [about this repo ](https://jupyter-accessibility.readthedocs.io/en/restructure/README.html) is on Read The Docs.

-----

File: CONTRIBUTING.md
Comment (bay1877) [reply CONTRIBUTING.md:25]: yes, they are supposed to be links. they are reference style links with the references on lines 178-179
```
[Jupyter Gitter]: https://gitter.im/jupyter/jupyter
[Jupyter Website]: https://jupyter.org/
```

File: CONTRIBUTING.md
Comment (bay1877) [reply CONTRIBUTING.md:36]: yes, similar to above. Reference style links with the references on line 180.
```
[JupyterLab Repository]:https://github.com/jupyterlab/jupyterlab
```

File: docs/community/README.md
Comment (bay1877) [reply docs/community/README.md:16]: i am linking to the all minutes markdown on line 18. Do we just not want the old archive at all anymore? or does the "all meeting minutes" include the "issues/98" meeting minutes too?
```
- [All JupyterLab Accessibility Meeting Minutes](./meeting-minutes/jupyterlab-accessibility-meetings/all-minutes.md)
```

File: docs/community/README.md
Comment (bay1877) [reply docs/community/README.md:17]: Looking back, I cant recall of any significant reasoning. Most likely just to separate the two bulleted lists for some reason. I am not opposed to combining them though. 

I think maybe my thought was the first bulleted list was more of the "meeting notes" and the second was more of the "reports". Although, the meeting minutes on line 18 would be in the wrong list under that logic.

**Comment by bay1877** [comment]
Okay, so I drafted a potential solution to avoid duplicated information. Certainly open to feedback on the approach, especially on preferred naming and directory/file placement.

My initial design/solution is the following...

As pointed out, some of the information between the README.md and the index.md (the new home page of the read the docs) was repeated/duplicated. Originally, the home page of the read the docs was just the README.md file, and I wasn't a huge fan of this for some of the same reasons (homepage covered more detail than needed/repeated sections like the "get involved section".

I broke down the README.md into three "chunk" markdown files for each of the sections, and I put them into `./docs/readme_chunks`. I made a bash script `./docs/readme_chunks/generate_frontpage_documents.sh` that took these chunks and generated the index.md and the README.md. Instead of having to modify information in two places, simply modify the right "chunk" and then run the script to generate both documents.

Pros vs Cons to this approach:
pros -> prevents duplicate information/changes
cons -> adding a new "chunk" to either now requires adding a new chunk md file and updating the generation script to concatenate it into the right place.

relavant commits:
https://github.com/jupyter/accessibility/pull/82/commits/2d793278dabc987fb8d4aab29c1aa6d9e07ffd10
https://github.com/jupyter/accessibility/pull/82/commits/a2b9ba937e168b841f606b7cb8003abe5a9a2524

**Comment by pws1453** [comment]
Made a python version of this. The code has internal references to existing chunks, but these references can be added to with ease. Additionally, external code can easily grab a list of chunk paths. To add a chunk to the script, the chunk path would be added as a variable, which would need to be added to the return value of `get_chunk_paths`. This file would also need to be opened and its lines read into a variable, as done in `__parse_files`. This code works well for the files that need to be written in this PR, and could be further modularised with ease (one of the nice things about OO)  in case this chunked approach was to be re-used in other contexts.

File: CONTRIBUTING.md
Comment (isabela-pf) [reply CONTRIBUTING.md:36]: Nope, links are good! I just was not familiar with the styling.

File: docs/community/README.md
Comment (isabela-pf) [reply docs/community/README.md:16]: `all-minutes.md` does have everything listed on jupyterlab/team-compass #98. I'd prefer to point people in one place, but I won't block it if you see benefit in keeping the old link too.

**Review by isabela-pf — APPROVED** [review]
Thanks for taking the feedback! This looks good to me, but I do not have the relevant skills to understand all of the changes. @gabalafou would you mind taking another look? I'll be happy to merge once I get your approval, too.

Also thanks for giving me a much needed and relatable laugh with the `maaaay have forgotten to add summary to docstring`commit message.

**Review by gabalafou — CHANGES_REQUESTED** [review]
Thanks @isabela-pf for pinging me on this PR. Apologies for not getting to it sooner! 

I soooo want to be able to accept this PR. However, unless I misunderstand how it works, I think it has a fatal flaw.

Let's say I want to update the following sentence in the introduction, changing the month April to May:

```diff
- As of April 2022, significant work is still required to reach that goal.
+ As of May 2022, significant work is still required to reach that goal.
```

Are these the steps that I would follow?

1. Checkout new branch in Github
2. Edit the sentence in `docs/readme_chunks/introduction.md`
3. Run your Python script
4. git add, commit the following three files, then push the branch to Github:
```
docs/readme_chunks/introduction.md
docs/index.md
./README.md
```

If that is correct, then the fatal flaw is that even though you're generating the docs, the fact that we have to check in the duplicated content into the repo will create problems for us. In other words, we need deduplication at the level of what's checked into the repo. 

Given that this repo is open source and community driven, I don't think we can afford to violate certain conventions and expectations. What I mean is, if a new person comes to this repo and they see markdown files in a docs section in a Github repo, the expectation is that if they want to update the docs, then they simply update the markdown file. That's how it works for almost every repo I've ever seen on Github. I feel that if we violate this convention, we are essentially putting a banana on the floor and hoping somebody doesn't slip on it.

The only workable solutions I can imagine right now are

- the one I mentioned in an earlier comment, simply deleting the duplicate content where it exists and pointing the reader to the single source of truth with a link
- if JupyterBook has a way to index off the repo's root level README, maybe that would work?

I just checked the [JupyterBook repo](https://github.com/executablebooks/jupyter-book) to see what they do. It looks like their Github README is completely separate from their JupyterBook content, so heh, I'm not sure if there's a way to do what we want. 

I would be happy to pair program with you to help finish and close out this PR. If you're interested, you can find me on Gitter (same username as the one I'm using here on Github) and we can coordinate from there.

File: .vscode/settings.json
Comment (trallard) [reply .vscode/settings.json:1]: I am going to make the assumption that the addition was unintended - can this be removed, please?

File: docs/developer/CONTRIBUTING_HERE.md
Comment (trallard) [inline docs/developer/CONTRIBUTING_HERE.md:3]: I am assuming your editor is wrapping text at a certain lenght - can you split this at 120 char? it makes it so much easier to review PRs 🙏🏼

File: docs/developer/CONTRIBUTING_HERE.md
Comment (trallard) [inline docs/developer/CONTRIBUTING_HERE.md:28]: Is this section intended to be empty? 
if there is no content it would be best to remove to avoid confusion

File: docs/readme_chunks/generate_frontpage_documents.py
Comment (trallard) [inline docs/readme_chunks/generate_frontpage_documents.py:17]: Dealing with paths can be hard - I would suggest using Pathlib to deal with all the paths-related operations and have absolute paths

File: docs/readme_chunks/generate_frontpage_documents.py
Comment (trallard) [inline docs/readme_chunks/generate_frontpage_documents.py:34]: Same from here onwards - I would suggest using Pathlib to leverage the use of `parent()` and the such

File: docs/readme_chunks/generate_frontpage_documents.py
Comment (trallard) [reply docs/readme_chunks/generate_frontpage_documents.py:86]: Right - would it make more sense to have a single method that iterates over a list or something to programmatically open and read the files?

File: docs/readme_chunks/generate_frontpage_documents.py
Comment (trallard) [inline docs/readme_chunks/generate_frontpage_documents.py:1]: I feel like I am missing some context here - why is the reading/chunking  needed to generate the docs?

File: CONTRIBUTING.md
Comment (trallard) [inline CONTRIBUTING.md:20]:
```suggestion
## Community Structure - The Jupyter Project and Subprojects

```

File: CONTRIBUTING.md
Comment (trallard) [inline CONTRIBUTING.md:31]:
```suggestion
### JupyterLab

```

File: CONTRIBUTING.md
Comment (trallard) [inline CONTRIBUTING.md:42]:
```suggestion
### JupyterNotebook

```

File: CONTRIBUTING.md
Comment (trallard) [inline CONTRIBUTING.md:49]:
```suggestion
### JupyterHub

```

File: CONTRIBUTING.md
Comment (trallard) [inline CONTRIBUTING.md:58]:
```suggestion
computing group.

```

File: RUN-DOCS.md
Comment (trallard) [inline RUN-DOCS.md:1]:
```suggestion
# How to build the docs locally

```
If I read correctly this is what we are aiming for

File: RUN-DOCS.md
Comment (trallard) [inline RUN-DOCS.md:6]: Could you add steps on what the contributor is expected to do later on?

**Review by trallard — CHANGES_REQUESTED** [review]

File: docs/readme_chunks/generate_frontpage_documents.py
Comment (pws1453) [reply docs/readme_chunks/generate_frontpage_documents.py:86]: I put everything in 'with' blocks to allow the files to automatically be closed. I will def look for a better solution.

**Comment by pws1453** [comment]
First - Sorry for the delays. Finals hit and I've just begun a new internship.

I think the best solution may be that we establish a 'single source of truth', as previously suggested by @gabalafou , whether that be the RTD or the Github readme, and link one to another. This would reduce developer confusion, and allows the community to avoid keeping up the chunking subsystem.

We could also have CI make a copy of the 'single source of truth' upon a new commit to master to wherever we need the source to be copied to. However, this is another thing to upkeep. I will definitely recommit without the .vscode files - missed those. Sorry.

**Comment by gabalafou** [comment]
@pws1453 congrats on the internship and thanks for following up!

I prefer the first solution you mention (linking) rather than using CI (because of upkeep, as you mention). 

Will you be able to update this PR?

**Comment by trallard** [comment]
Since this has been open for a while with no movement - I will close this PR as #106  tackles many of the review issues.

I will however add @pws1453  as a co-author in #106 since I used some of the structure and other work here to move the other PR 🙏🏽

**Pull request closed without merging**
